src/test/javascript/grouping.controller.test.js

```diff
@@ -918,6 +920,42 @@ describe("GroupingController", () => {
             expect(scope.isBatchImport).toBeTrue();
         });
 
+        it("should not update variables within $scope.addMembers if $scope.isAddingMembers is true", () => {
+            scope.isAddingMembers = true;
+            scope.addMembers("Include", uhIdentifiers);
+
+            expect(scope.listName).toBeUndefined();
+            expect(scope.errorDismissed).toBeUndefined();
+            expect(scope.isBatchImport).toBeUndefined();
+            expect(scope.containsInput).toBeUndefined();
+            expect(scope.invalidMembers).toEqual([]);
+            expect(scope.addInputError).toBeUndefined();
+            expect(scope.waitingForImportResponse).toBeFalse();
+        });
+
+        it("should not call functions within $scope.addMembers if $scope.isAddingMembers is true", () => {
+            scope.isAddingMembers = true;
+            scope.addMembers("Include", uhIdentifiers);
+
+            spyOn(scope, 'displayDynamicModal');
+            spyOn(scope, 'existsInList');
+            spyOn(gs, 'getMemberAttributeResultsAsync');
+            spyOn(gs, 'getMemberAttributeResults');
+            spyOn(scope, 'displayImportErrorModal');
+            spyOn(scope, 'displayImportConfirmationModal');
+            spyOn(scope, 'displayAddModal');
+            spyOn(scope, 'displayApiErrorModal');
+
+            expect(scope.displayDynamicModal).not.toHaveBeenCalled();
+            expect(scope.existsInList).not.toHaveBeenCalled();
+            expect(gs.getMemberAttributeResultsAsync).not.toHaveBeenCalled();
+            expect(gs.getMemberAttributeResults).not.toHaveBeenCalled();
+            expect(scope.displayImportErrorModal).not.toHaveBeenCalled();
+            expect(scope.displayImportConfirmationModal).not.toHaveBeenCalled();
+            expect(scope.displayAddModal).not.toHaveBeenCalled();
+            expect(scope.displayApiErrorModal).not.toHaveBeenCalled();
```

Comment:
Sorry I should have been more specific, I would only check that most important function was not called. In this case, it would be `gs.getMemberAttributeResultsAsync` and `gs.getMemberAttributeResults` becuase it invokes an API call. A problem we have with our unit tests is that they are extremely fragile where editing one line of code just breaks a bunch of tests. Just keeping it minimal here is the best approach.